flang/lib/Lower/Bridge.cpp

```diff
@@ -2078,33 +2084,35 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     // bounds from the caller (boxed somewhere else). Locals have the same
     // properties except they are never boxed arguments from the caller and
     // never having a missing column size.
-    mlir::Value addr = lookupSymbol(sym);
+
+    // Arguments (and some results) already have a symbolBox with the address.
+    auto maybeSymbolBox = lookupSymbol(sym);
+    mlir::Value addr =
+        maybeSymbolBox ? maybeSymbolBox->getAddr() : mlir::Value{};
     mlir::Value len;
     [[maybe_unused]] bool mustBeDummy = false;
 
     if (sba.isChar) {
       // if element type is a CHARACTER, determine the LEN value
       if (isDummy || isResult) {
         auto unboxchar = charHelp.createUnboxChar(addr);
-        auto boxAddr = unboxchar.first;
+        addr = unboxchar.first;
```

Comment:
Please try to keep changes here to a minimum as I'm refactoring this code now.

Reply:
OK, but I absolutely need to not propagate the boxchar anymore here, otherwise the rest of the refactoring won't work.
I will not change anything more in this function though.

Reply:
Yup. Thank you.